src/modules/navigator/navigator_main.cpp

```diff
@@ -774,7 +788,37 @@ Navigator::run()
 	}
 }
 
-void Navigator::geofence_breach_check(bool &have_geofence_position_data)
+bool Navigator::geofence_breach_check_while_on_ground(bool &have_geofence_position_data)
```

Comment:
Could the existing geofence not do the calculations already?

Reply:
@MaEtUgR No, it can't. If you have a look at the logic you will see that the violation test point rotates around the vehicle which is an attempt to get the distance to the fence. If you are flying then something similar happens but the point is predicted using the velocity vector of the vehicle.
You can of course push all this into the geofence such that you remain with a function called "update" in navigator but I'm not convinced this helps.

Reply:
> you will see that the violation test point rotates around the vehicle

I saw that. Then it's a problem with how the geofence check is currently implemented. It would be much nicer to have a library call to check a radius without rotating a vector around arbitrary angles to mimic a non-existing velocity explicitly just for the case the vehicle is on the ground.